dispatches/models/renewables_case/elec_splitter.py

```diff
@@ -163,39 +187,59 @@ def initialize(self, **kwargs):
+                if n == len(self.outlet_list) - 1:
+                    continue
+                # if too many are fixed, unfix the first, generally assume that is
+                # the main flow, and is the calculated split fraction
+                elif n == len(self.outlet_list):
+                    self.split_fraction[self.outlet_list[0], t].unfix()
+                # if not enough fixed, start fixing from the back until there are
+                # are enough
+                else:
+                    for o in reversed(self.outlet_list):
+                        if not self.split_fraction[o, t].fixed:
+                            self.split_fraction[o, t].fix()
+                            n += 1
+                        if n == len(self.outlet_list) - 1:
+                            break
+            self.electricity.fix()
             for o in self.outlet_list:
-                elec_obj = getattr(self, o + "_elec")
-                if elec_obj[t].fixed:
-                    self.split_fraction[o, t].fix(
-                        value(elec_obj[t] / self.electricity[t]))
-
-        # fix or unfix split fractions so n - 1 are fixed
-        for t in self.flowsheet().config.time:
-            # see how many split fractions are fixed
-            n = sum(1 for o in self.outlet_list if self.split_fraction[o, t].fixed)
-            # if number of outlets - 1 we're good
-            if n == len(self.outlet_list) - 1:
-                continue
-            # if too many are fixed, unfix the first, generally assume that is
-            # the main flow, and is the calculated split fraction
-            elif n == len(self.outlet_list):
-                self.split_fraction[self.outlet_list[0], t].unfix()
-            # if not enough fixed, start fixing from the back until there are
-            # are enough
-            else:
-                for o in reversed(self.outlet_list):
-                    if not self.split_fraction[o, t].fixed:
-                        self.split_fraction[o, t].fix()
-                        n += 1
-                    if n == len(self.outlet_list) - 1:
-                        break
-
-        self.electricity.fix()
-        for o in self.outlet_list:
-            getattr(self, o + "_port").unfix()
+                getattr(self, o + "_port").unfix()
+        else:
+            # fix or unfix electricity flows so n - 1 are fixed
+            for t in self.flowsheet().config.time:
+                # see how many electricity flows are fixed
+                n = sum(1 for o in self.outlet_list if getattr(self, o + "_elec")[t].fixed)
+                # if number of outlets - 1 we're good
+                if n == len(self.outlet_list) - 1:
+                    continue
+                # if too many are fixed, unfix the first, generally assume that is the main flow
+                elif n == len(self.outlet_list):
+                    getattr(self, self.outlet_list[0] + "_elec")[t].unfix()
```

Comment:
Here, we are changing the degrees of freedom by fixing/unfixing variables to make it a square problem. However, those variables are not reverted to their original state after the initialization. Either we need to print a warning saying that the state of the variables has been changed (so that the users are aware of it), or we need to get it back to their original state.

Reply:
Doesn't the `to_json` in line 187 and the `from_json` in line 251 handle restoring the original state?

Reply:
Sorry, Darice! I missed those two lines. I'm not familiar with all the options of `StoreSpec` class. In line 187, you are storing the list of variables that are fixed. Does the dictionary `istate` contain the variables that are not fixed? 
In line 251, you are passing this dictionary `istate` as input. I can see that it reverts the variables which are fixed by the user to their initial state. But how about the variables which were not fixed by the user but fixed by the initialization method? I'm assuming that the information about these variables will not be in the `istate` dictionary, so `to_json` function does not unfix these variables. Please correct me if my understanding is not correct.

Reply:
No problem! I have checked those functions and even with `only_fixed=True`, the `from_json(self, sd=istate, wts=sp)` does restore previously unfixed variables back to unfixed state if the initialization fixes them. The DOF before and after initialization is the same in all the tests I have for elec_splitter.py.

Reply:
Alright. Thank you, Darice, for clarifying the clarification!